alphastats/gui/utils/preprocessing_helper.py

```diff
@@ -1,124 +1,225 @@
 import streamlit as st
 import pandas as pd
+from st_cytoscape import cytoscape
 
 import datetime
 
+CYTOSCAPE_STYLESHEET = [
+    {
+        "selector": "node",
+        "style": {
+            "label": "data(label)",
+            "shape": "roundrectangle",
+            "width": 200,
+            "height": 60,
+            "text-valign": "center",
+            "text-halign": "center",
+        },
+    },
+    {
+        "selector": "node.active",
+        "style": {
+            "background-color": "lightgreen",
+        },
+    },
+    {
+        "selector": "node.inactive",
+        "style": {
+            "background-color": "lightgrey",
+        },
+    },
+    {
+        "selector": "node.selected",  # TODO: This currently does not work, figure out why
+        "style": {"background-color": "red"},
+    },
+    {
+        "selector": "edge",
+        "style": {
+            "width": 3,
+            "curve-style": "bezier",
+            "target-arrow-shape": "triangle",
+        },
+    },
+]
+
+# TODO: Make help texts meaningful
+# TODO: Show help texts on the widgets
+WORKFLOW_STEPS = {
+    "remove_contaminations": {
+        "repr": "Remove contaminations",
+        "help": "Remove contaminations annotated in the contaminations library and filter columns included in the dataset.",
+    },
+    "remove_samples": {
+        "repr": "Remove samples",
+        "help": "Remove samples from analysis, e.g. useful when failed or blank runs are included.",
+    },
+    "subset": {
+        "repr": "Subset data",
+        "help": "Subset data so it matches with metadata. Can for example be useful if several dimensions of an experiment were analysed together.",
+    },
+    "data_completeness": {
+        "repr": "Filter data completeness",
+        "help": "Filter data based on completeness across samples. E.g. if a protein has to be detected in at least 70% of the samples.",
+    },
+    "log2_transform": {"repr": "Log2 transform", "help": "Log2-transform dataset."},
+    "normalization": {
+        "repr": "Normalization",
+        "help": 'Normalize data using one of the available methods ("zscore", "quantile", "vst", "linear").',
+    },
+    "imputation": {
+        "repr": "Imputation",
+        "help": 'Impute missing values using one of the available methods ("mean", "median", "knn", "randomforest").',
+    },
+    "batch": {"repr": "Batch correction", "help": "Batch correction."},
+}
+
+PREDEFINED_ORDER = [
+    "remove_contaminations",
+    "remove_samples",
+    "subset",
+    "data_completeness",
+    "log2_transform",
+    "normalization",
+    "imputation",
+    "batch",
+]
+
+
+def draw_workflow(workflow: list[str], order: list[str] = PREDEFINED_ORDER):
```

Comment:
could move this and `CYTOSCAPE_STYLESHEET` to a new module `preprocessing_drawer` or smth

Reply:
I agree on the stylesheet, but not the function. I would leave it as is now unless it really bothers you.